lib/awful/spawn.lua

```diff
@@ -318,21 +318,75 @@ function spawn.with_line_callback(cmd, callbacks)
     return pid
 end
 
---- Asynchronously spawn a program and capture its output.
--- (wraps `spawn.with_line_callback`).
+--- Asynchronously spawn a program and (optionally) execute callbacks.
+--
+-- `cmd` is being run and in case `cb`, `cb_error` or `cb_signal` are provided
+-- they will be run, according to the exit code of `cmd`.
+--
+-- This wraps `spawn.with_line_callback`.
+--
 -- @tparam string|table cmd The command.
--- @tab callback Function with the following arguments
--- @tparam string callback.stdout Output on stdout.
--- @tparam string callback.stderr Output on stderr.
--- @tparam string callback.exitreason Exit Reason.
```

Comment:
If this was removed, then the API is broken and this can't be merged until v5 is branched. Please make sure the API remains compatible. Feel free to add `--TODO v5 deprecate this`